cluster-autoscaler/core/scale_down.go

```diff
@@ -1086,35 +1083,34 @@ func (sd *ScaleDown) getEmptyNodes(candidates []string, maxEmptyBulkDelete int,
 			}
 			available--
 			availabilityMap[nodeGroup.Id()] = available
-			result = append(result, node)
+			nodesToRemove = append(nodesToRemove, simulator.NodeToBeRemoved{
+				Node: node,
+			})
 		}
 	}
-	limit := maxEmptyBulkDelete
-	if len(result) < limit {
-		limit = len(result)
-	}
-	return result[:limit]
+
+	return sd.processors.ScaleDownSetProcessor.GetNodesToRemove(nodesToRemove, maxEmptyBulkDelete)
```

Comment:
Why does this method call the processor inside itself, but `FindNodesToRemove` doesn't? If the purpose of the processor is to pick final scale-down candidates, I'd expect it to be called in exactly two places - ~right before `scheduleDeleteEmptyNodes`, and ~right before `deleteNode` in `TryToScaleDown`.

Reply:
Moved the calls to the `GetNodesToRemove` just before the actual deletion.

Reply:
Thanks, reads much better now!